great_expectations/datasource/fluent/sources.py

```diff
@@ -273,28 +273,21 @@ def factories(self) -> List[str]:
     def __getattr__(self, attr_name: str):
         try:
             ds_constructor = self.__source_factories[attr_name]
-
-            def wrapped(name: str, **kwargs):
-                datasource = ds_constructor(name=name, **kwargs)
-                # using try/except because hasattr will fail to find pydantic PrivateAttrs
-                try:
-                    datasource._data_context = self._data_context
-                except ValueError:
-                    logger.debug(
-                        f'Failed to attach data context to datasource "{name}" because {datasource} has no attribute "_data_context".'
-                    )
-                    pass
-                # TODO (bdirks): _attach_datasource_to_context to the AbstractDataContext class
-                self._data_context._attach_datasource_to_context(datasource)
-                return datasource
-
-            wrapped.__doc__ = ds_constructor.__doc__
-            # attr-defined issue https://github.com/python/mypy/issues/12472
-            wrapped.__signature__ = ds_constructor.__signature__  # type: ignore[attr-defined]
-            return wrapped
         except KeyError:
             raise AttributeError(f"No factory {attr_name} in {self.factories}")
 
+        def wrapped(name: str, **kwargs):
+            datasource = ds_constructor(name=name, **kwargs)
+            datasource._data_context = self._data_context
+            # TODO (bdirks): _attach_datasource_to_context to the AbstractDataContext class
+            self._data_context._attach_datasource_to_context(datasource)
```

Comment:
Try/except removed from here now that `_data_context` exists on all `Datasource`s.

Reply:
Sorry actually on line 281